src/runtime/methodbinder.cs

```diff
@@ -600,10 +614,11 @@ static IntPtr HandleParamsArray(IntPtr args, int arrayStart, int pyArgCount, out
         /// <param name="op">Pointer to the object at a particular parameter.</param>
         /// <param name="parameterType">That parameter's managed type.</param>
         /// <param name="needsResolution">There are multiple overloading methods that need resolution.</param>
+        /// <param name="setError">If true, call <c>Exceptions.SetError</c> with the reason for failure.</param>
         /// <param name="arg">Converted argument.</param>
         /// <param name="isOut">Whether the CLR type is passed by reference.</param>
         /// <returns></returns>
-        static bool TryConvertArgument(IntPtr op, Type parameterType, bool needsResolution,
+        static bool TryConvertArgument(IntPtr op, Type parameterType, bool needsResolution, bool setError,
```

Comment:
Same here: is there a reason to ever pass `setError: false`?

Reply:
I can see two reasons:
1. If there were > 1 overload, and none matched, you would only get the last error.
2. If there were 10 overloads, and the last matched, you would waste time constructing error messages.
If you are happy with these consequences then I will make setError always true.

Reply:
1. Can you make in 1. so that all errors are kept as `AggregateException`?
2. I think that is fine. We might need a guide suggesting to choose overload if they need performance anyway. Because time is always wasted trying to convert parameters.